src/xml.ts

```diff
@@ -121,6 +122,17 @@ export const signXml = (
   if (options.signatureAlgorithm != null) {
     sig.signatureAlgorithm = algorithms.getSigningAlgorithm(options.signatureAlgorithm);
   }
+  if (options.signingCert != null) {
+    const certArray = Array.isArray(options.signingCert)
+      ? options.signingCert
+      : [options.signingCert];
+    const cert = removeCertPEMHeaderAndFooter(certArray[0]);
```

Comment:
I'm using existing function here - "removeCertPEMHeaderAndFooter" - but this failing my UT for polynomial regular expression.. as shown below.. Any suggestions on this..?
![image](https://user-images.githubusercontent.com/6399781/162631832-9ae08abd-12d4-42ce-8327-7dcf32b1ba8a.png)

Reply:
Interesting find. The problem is with these lines:

```TypeScript
export const removeCertPEMHeaderAndFooter = (certificate: string): string => {
  certificate = certificate.replace(/-+BEGIN CERTIFICATE-+\r?\n?/, "");
  certificate = certificate.replace(/-+END CERTIFICATE-+\r?\n?/, "");
  certificate = certificate.replace(/\r\n/g, "\n");
  return certificate;
};
```

Specifically the `-+` part. There is a suggestion on how to fix that [here](https://codeql.github.com/codeql-query-help/javascript/js-polynomial-redos/). Feel free to open a PR to address this. Something as simple as this can even land before this, and very quickly at that.

Reply:
This has been resolved by https://github.com/node-saml/node-saml/pull/79. Please merge `master` in to your branch and have a look at code coverage.

Reply:
I just did the merge for you. Have a look.